src/main/java/com/google/codeu/data/Datastore.java

```diff
@@ -145,4 +146,53 @@ public User getUser(String email) {
     
     return user;
   }
+  /** Fetches markers from Datastore. */
+  public List<Marker> getMarkers() {
+    List<Marker> markers = new ArrayList<>();
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    Query query = new Query("Marker");
+    PreparedQuery results = datastore.prepare(query);
+
+    for (Entity entity : results.asIterable()) {
+      double lat = (double) entity.getProperty("lat");
+      double lng = (double) entity.getProperty("lng");
+      String content = (String) entity.getProperty("content");
+
+      Marker marker = new Marker(lat, lng, content);
+      markers.add(marker);
+    }
+    return markers;
+  }
+
+  /** Stores a marker in Datastore. */
+  public void storeMarker(Marker marker) {
+    Entity markerEntity = new Entity("Marker");
+    markerEntity.setProperty("lat", marker.getLat());
+    markerEntity.setProperty("lng", marker.getLng());
+    markerEntity.setProperty("content", marker.getContent());
+
+    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
+    datastore.put(markerEntity);
+  }
+
+  public Marker removeMarker(Marker marker) {
+    Query query = new Query("Marker")
+      .setFilter(CompositeFilterOperator.and(
+        new Query.FilterPredicate("lat", FilterOperator.EQUAL, marker.getLat()),
+        new Query.FilterPredicate("lng", FilterOperator.EQUAL, marker.getLng())));
```

Comment:
I used only lat and lng because in any case, it is very difficult to place a marker in the exact same coordinates do to them having a lot of decimals. Also using only content as an identifier could end up in deleting markers in completely different positions but with the same text.

Another option would be to use UUIDs like in the Message class, but I thought it was unecessary due to the precision of the lat/lng coordinates

Reply:
I mean when you are filtering, you are just checking for lat and lng to be the same, but the content field can be different. Would two markers with the same lat and lng, but different content, be considered the same marker? It seems like that's the case here

Reply:
I guess it would, but it shouldn't since it could be an unexpected behavior in the user's eyes. I'll add the content to match as well.